app/controllers/api/v1x0/settings_controller.rb

```diff
@@ -0,0 +1,41 @@
+module Api
+  module V1x0
+    class SettingsController < ApplicationController
+      before_action :org_admin_check
+
+      def index
+        settings = Catalog::TenantSettings.new(tenant)
+        render :json => settings.process.response
+      end
+
+      def show
+        render :json => { params.require(:id) => setting(params.require(:id)) }
+      end
+
+      def create
+        tenant.add_setting(params.require(:name), params.require(:value))
+        render :json => { params[:name] => setting(params[:name]) }
+      end
+
+      def update
+        tenant.update_setting(params.require(:id), params.require(:value))
+        render :json => { params[:id] => setting(params[:id]) }
+      end
+
+      def destroy
+        tenant.delete_setting(params.require(:id))
+        head :no_content
+      end
+
+      private
+
+      def setting(name)
+        tenant.settings[name]
+      end
+
+      def tenant
```

Comment:
I wonder if this variable should get set in the same `before_filter` that validates if the user has the correct permissions?